### DISCUSSION TITLE
Vercel + Next.js on production | API - data fetch

### DISCUSSION THREAD

**Discussion by enkicoma** [vercel/next.js#22634] [Help]
Hi 

I have described my issue here, hope someone can explain ^_^

https://stackoverflow.com/q/66420895/8780756

**Comment by michrome** [comment]
You should not use `fetch()` to call your own API routes server-side. Instead, directly import the logic used inside your API route. You may need to slightly refactor your code for this approach.

More info at https://nextjs.org/docs/basic-features/data-fetching

It's best not to read your own `/api` endpoints server side. Instead, : they're really designed to

**Reply by enkicoma** [reply]
Hmm a bit confused when you said not to fetch and call my API. I don't know the best practices, but I will try to explain my story, hope that will clarify a bit.

I created a next /api/notes with the idea for a POST, PUT, DELETE, GET request with the JSON data available on that path. 
<img width="215" alt="Screenshot 2021-03-01 at 18 18 04" src="https://user-images.githubusercontent.com/29044574/109540518-7a579d00-7aba-11eb-8a65-57678d7d1236.png">

the link you provided has `fetch` everywhere, what should be the value for production 
```const res = await fetch(https://.../data)``` res variable? (My DNS?)

**Reply by michrome** [reply]
Don't fetch your own APIs for anything you're doing that's already server-side e.g `getStaticProps` or `getServerSideProps`.

You fetch other services' APIs because that's all they expose to you: an HTTP end point.

But you have your own source code … so whatever you're doing in your API can be extracted into a local library. You can use it directly, without needing to go via HTTP.
